pipeline/authn/authenticator_cookie_session.go

```diff
@@ -115,10 +115,12 @@ func forwardRequestToSessionStore(r *http.Request, checkSessionURL string) (json
 	if res.StatusCode == 200 {
 		body, err := ioutil.ReadAll(res.Body)
 		if err != nil {
-			return json.RawMessage{}, err
+			return json.RawMessage{}, helper.ErrForbidden.WithReason(err.Error()).WithTrace(err)
```

Comment:
- Why would you consider the inability to read the response body a 401 error? Wouldn't it make more sense to return more error details here? I believe this will be incredibly hard to handle. I understand that this was the case before, but since we're working on this patch we can address it straight away.
- `.WithTrace()` adds the stack trace from the upstream error to the error context. stdlib errors do not implement stack trace functionality, therefore `WithTrace()` on a stdlib doesn't have an effect.
- Without `errors.WithStack()` it's not possible to get the error trace. What was your thinking behind omitting that?

Reply:
I added that here because the function that called `forwardRequestToSessionStore` was previously wrapping all the errors in a 403, so I added this to maintain the existing behavior for all of the other error cases, though I would say this should probably be a 500 error instead

Reply:
Ok, so let's return instead:

```suggestion
			return json.RawMessage{}, errors.WithStack(helper.ErrForbidden.WithReasonf("Unable to fetch cookie session context from remote: %+v", err))
```